docs/source/x_ext.rst

```diff
@@ -1287,19 +1299,19 @@ Handshake dependencies
 In order to avoid system level deadlock both the |processor| and the |coprocessor| shall obey the following rules:
 
 * The ``valid`` signal of a transaction shall not be dependent on the corresponding ``ready`` signal.
-* Transactions related to an earlier part of the instruction flow shall not depend on transactions with the same ``hartid`` and ``id`` related to a later part of the instruction flow. The instruction flow is defined from earlier to later as follows:
+* The only allowed dependencies between interfaces for transactions with the same ``hartid`` and ``id``are:
 
-  * compressed transaction
-  * issue transaction
-  * register transaction
-  * commit transaction
+  * ``issue_req`` may depend on ``compressed_resp``
+  * ``register`` may depend on ``compressed_resp``, ``issue_resp``, and ``commit``
+  * ``commit`` may depend on ``issue_resp``
```

Comment:
Why not ``commit`` may depend on ``issue_resp`` and ``register``?

Reply:
what information from register should drive a commit decision?
The dependencies need to be a directed graph without any loops to guarantee deadlock freedom. so if register is allowed to depend on commit, commit may not depend on register

Reply:
thinking on this, register does not need to rely on commit at all. I fixed it in 3f77c48.
Reason being is that both commit and register and CPU signals. Thus, i case of a commit-kill condition, the CPU will deassert register_valid anyway. This is not a dependence of register on commit.